### ISSUE TITLE
🐛 [tailwind] Missing semicolon after @apply breaks parsing

### ISSUE THREAD

**Issue by Kakadus** [biomejs/biome#8715]
### Environment information

<details>

```bash
biome rage
CLI:
  Version:                      2.3.11
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v22.21.1
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         unset

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.json
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  true
  HTML full support enabled:    unset

Workspace:
  Open Documents:               0
```

</details>

### What happened?

1. Have an `@apply` as the last statement inside a block (e.g. `@utility container`)
2. Do not terminate the statement with `;`
3. See the error `expected ; but instead the file ends` but AFTER the closing bracket of the block. Prettier is able to correctly format this.

e.g. 
```css
@utility container {
  @apply mx-auto space-y-4 px-8 py-10 sm:px-16 md:px-24 lg:px-32
}
```
See https://next.biomejs.dev/playground/?cssModules=true&code=QAB1AHQAaQBsAGkAdAB5ACAAYwBvAG4AdABhAGkAbgBlAHIAIAB7AAoAIAAgAEAAYQBwAHAAbAB5ACAAbQB4AC0AYQB1AHQAbwAgAHMAcABhAGMAZQAtAHkALQA0ACAAcAB4AC0AOAAgAHAAeQAtADEAMAAgAHMAbQA6AHAAeAAtADEANgAgAG0AZAA6AHAAeAAtADIANAAgAGwAZwA6AHAAeAAtADMAMgAKAH0ACgA%3D&language=css

If I were to manually "correct" the errors highlighted by the biome parser, biome would accept the input, but no longer prettier, because there would be more closing brackets than opening ones. It seems like biome mangles the closing bracket of the block into the `@apply` statement.
e.g.
```css
@utility container {
  @apply mx-auto space-y-4 px-8 py-10 sm:px-16 md:px-24 lg:px-32
}
;}
```
See https://next.biomejs.dev/playground/?cssModules=true&code=QAB1AHQAaQBsAGkAdAB5ACAAYwBvAG4AdABhAGkAbgBlAHIAIAB7AAoAIAAgAEAAYQBwAHAAbAB5ACAAbQB4AC0AYQB1AHQAbwAgAHMAcABhAGMAZQAtAHkALQA0ACAAcAB4AC0AOAAgAHAAeQAtADEAMAAgAHMAbQA6AHAAeAAtADEANgAgAG0AZAA6AHAAeAAtADIANAAgAGwAZwA6AHAAeAAtADMAMgAKAH0ACgA7AH0A&language=css

### Expected result

In both cases, prettier seems to be right:

```css
@utility container {
	@apply mx-auto space-y-4 px-8 py-10 sm:px-16 md:px-24 lg:px-32;
}
```
and
```css
CssSyntaxError: Unexpected } (4:2)
  2 |   @apply mx-auto space-y-4 px-8 py-10 sm:px-16 md:px-24 lg:px-32
  3 | }
> 4 | ;}
    |  ^
```

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct
